ivy/functional/frontends/paddle/tensor/tensor.py

```diff
@@ -442,6 +442,12 @@ def floor_divide(self, y, name=None):
         y_ivy = y._ivy_array if isinstance(y, Tensor) else _to_ivy_array(y)
         return ivy.floor_divide(self._ivy_array, y_ivy)
 
+    @with_supported_dtypes({"2.5.0 and below": ("int32", "int64")}, "paddle")

```

Comment:
Where did you get the `supported_dtypes` of `int32, int64`? It seems like they have an example [in their docs](https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/bincount_en.html) that works for float as well:
```python
w = paddle.to_tensor([2.1, 0.4, 0.1, 0.5, 0.5])
result2 = paddle.bincount(x, weights=w)
print(result2) # [0., 2.19999981, 0.40000001, 0., 0.50000000, 0.50000000]
```
A general rule of thumb is don't blindly copy what other functions do, but try to understand why that specific line of code is needed. If you're copying from the function right above (`floor_divide`), it's because they directly told us [in `floor_divide` docs](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/floor_divide_cn.html#canshu) that it only supports `int32, int64` input.

Reply:
Hi @xoiga123, apologies as I misunderstood how the `supported_dtypes` decorator works and assumed it did not apply to all the arguments and only the input tensor. In the example, you show above, the float dtype is of the weights. This is the source function, and I took the dtypes from here.
```py
def bincount(x, weights=None, minlength=0, name=None):
    """
    Computes frequency of each value in the input tensor.

    Args:
        x (Tensor): A Tensor with non-negative integer. Should be 1-D tensor.
        weights (Tensor, optional): Weight for each value in the input tensor. Should have the same shape as input. Default is None.
        minlength (int, optional): Minimum number of bins. Should be non-negative integer. Default is 0.
        name(str, optional): The default value is None.  Normally there is no need for user to set this
            property.  For more information, please refer to :ref:`api_guide_Name`.

    Returns:
        Tensor: The tensor of frequency.
    """
    if x.dtype not in [paddle.int32, paddle.int64]:
        raise TypeError("Elements in Input(x) should all be integers")

   ...
    check_variable_and_dtype(x, 'X', ['int32', 'int64'], 'bincount')

...
    if weights is not None:
        check_variable_and_dtype(
            weights,
            'Weights',
            ['int32', 'int64', 'float32', 'float64'],
            'bincount',
        )
   ...
``` and I unfortunately did not consider the types of weights.

I'll update these now

Reply:
@druvdub Ah you're entirely right actually, I didn't realize the float was for `weight`. This is correct.